### DISCUSSION TITLE
ESP32 ProS3 neopixel port

### DISCUSSION THREAD

**Discussion by hozmaster** [micropython/micropython#15232] [ESP32]
Micropython 1.2.3 and ProS3 board.

What is the port number to onboard RGB led ? 

```
from machine import Pin
from neopixel import NeoPixel

pin = Pin(0, Pin.OUT)    
np = NeoPixel(pin, 8)   
np[0] = (255, 255, 255) # set the first pixel to white
np.write()              # write data to all pixels
r, g, b = np[0]  
```
It should be in LD0 or LDO port based on CircuitPython example.

Or is there existing package which can be used directly ?

**Comment by hozmaster** [comment]
Most likely it exist in the port 17

https://github.com/UnexpectedMaker/esp32s3/blob/main/code/micropython/helper%20libraries/pros3/pros3.py

**Comment by bixb922** [comment]
The schematics show the neopixel led at GPIO18. ```p = Pin(18)``` should work in the code you posted. 

https://github.com/UnexpectedMaker/esp32s3/blob/main/schematics/schematic-pros3.pdf

**Comment by glenn20** [comment]
Yes, the neopixel is on pin 18 (RGB_DATA). Note that the neopixel requires that LDO2 (pin 17) is turned on to provide power to the LED.

You can find the common pin assignments in the ["pros3.py"](https://github.com/micropython/micropython/blob/master/ports/esp32/boards/UM_PROS3/modules/pros3.py) helper module which is frozen into the firmware image (and also in  ["pins.csv"](https://github.com/micropython/micropython/blob/master/ports/esp32/boards/UM_PROS3/pins.csv)).

eg.
```py
from machine import Pin
from neopixel import NeoPixel
from pros3 import RGB_DATA, set_ldo2_power

set_ldo2_power(1)   # Confirm the led is powered
pin = Pin(pros3.RGB_DATA, Pin.OUT)    
np = NeoPixel(pin, 8)   
np[0] = (255, 255, 255) # set the first pixel to white
np.write()              # write data to all pixels
r, g, b = np[0]  
```
